src/peakrdl_python/templates/addrmap.py.jinja

```diff
@@ -155,6 +163,34 @@ class {{get_fully_qualified_type_name(node)}}_cls(AddressMap):
         return self.__{{child_node.inst_name}}
             {%- endif %}
         {% endfor %}
+
+    {{ child_getter(node, "get_registers",   "Reg",        systemrdlRegNode) }}
+    {{ child_getter(node, "get_memories",    "Memory",     systemrdlMemNode) }}
+    {{ child_getter(node, "get_addressmaps", "AddressMap", systemrdlAddrmapNode) }}
+    {{ child_getter(node, "get_regfiles",    "RegFile",    systemrdlRegfileNode) }}
```

Comment:
I've not got my head around what an RDL regfile is for, and why I'd choose to use one over an addrmap. So I tend to forget about them. I'm happy to merge into 'sections'.

Reply:
The SystemRDL specification gives a clue:

> 13.1 Introduction
During generation, the address map can be converted into an HDL module. All registers and fields
instantiated within an address map file shall be generated within this module. Therefore, some properties are
only valid for addrmaps and not for regfiles. Other than these properties and their suggested behavior, there
is no difference between address maps and register files.

My understanding is you use AddrMap to guide any generator that it may put things into separate modules, whereas RegFile grouping concept that doesn't infer separation of the generated outputs. At the level of the PythonAPI (once which is a generated output) these two concepts can be merged.

Reply:
Thanks, that makes sense, and I can see where I'd want to use them now.
Yes, let's have a single iterator.

Reply:
This was quite easy to do with the macro you already provided:

Address Map Class changed to:
```
{{ child_getter(node, "get_sections", "Union[AddressMap, RegFile]", (systemrdlAddrmapNode, systemrdlRegfileNode)) }}
```

RegFile Class changed to: 
```
{{ child_getter(node, "get_sections",  "RegFile", systemrdlRegfileNode) }}
```